streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -91,6 +104,225 @@ public InternalTopicManager(final Time time,
         }
     }
 
+    static class ValidationResult {
+        public Set<String> missingTopics = new HashSet<>();
+        public Map<String, List<String>> misconfigurationsForTopics = new HashMap<>();
+    }
+
+    public ValidationResult validate(final Map<String, InternalTopicConfig> topicConfigs) {
```

Comment:
add a javadoc explaining what this method is doing and what the return value is?

Reply:
Will do!